src/main/java/org/opentripplanner/middleware/models/ItineraryExistence.java

```diff
@@ -0,0 +1,255 @@
+            if (result == null) {
+                result = new ItineraryExistenceResult();
+                setResultForDayOfWeek(result, dayOfWeek);
+            }
+            // Send off each plan query to OTP.
+            OtpDispatcherResponse response = OtpDispatcher.sendOtpPlanRequest(otpRequest);
+            TripPlan plan = response.getResponse().plan;
+            // Handle response if valid itineraries exist.
+            if (plan != null && plan.itineraries != null) {
+                for (Itinerary itineraryCandidate : plan.itineraries) {
+                    // If a matching itinerary is found, save the date with the matching itinerary.
+                    // The matching itinerary will replace the original trip.itinerary.
+                    // FIXME Replace 'equals' with matching itinerary
+                    if (itineraryCandidate.equals(referenceItinerary)) {
+                        result.handleValidDate(otpRequest.date, itineraryCandidate);
+                        hasMatchingItinerary = true;
+                    }
+                }
+            }
+            if (!hasMatchingItinerary) {
+                // If no match was found for the date, mark day of week as non-existent for the itinerary.
+                result.handleInvalidDate(otpRequest.date);
+            }
+        }
+        if (!this.allCheckedDaysAreValid()) {
+            this.message = String.format(
+                "The trip is not possible on the following days of the week you have selected: %s",
+                String.join(", ", getInvalidDaysOfWeek())
+            );
+            this.error = true;
+        }
+    }
+
+    /**
+     * Holds results for an itinerary existence check for a particular day of the week.
+     */
+    public static class ItineraryExistenceResult {
+        /**
+         * True if and only if an itinerary is available for all dates tested for existence.
+         */
+        @JsonProperty
+        public boolean isValid() {
+            return invalidDates.size() == 0;
+        }
+
+        /**
+         * Method and argument solely needed for accepting request contents that include a `valid` field
+         * (e.g. for checking a monitored trip with previous existence check record).
+         */
+        @JsonIgnore
+        public void setValid (boolean value) {}
+
+        /**
+         * Holds a list of invalid dates an itinerary is not available for the associated day of the week.
+         */
+        public List<String> invalidDates = new ArrayList<>();
+        /**
+         * Holds a list of valid dates for which an itinerary exists.
+         */
+        public List<String> validDates = new ArrayList<>();
+
+        /**
+         * Holds any matching itineraries (sorted by date) for the applicable day of the week.
+         * TODO: remove itineraries from list of returned fields in swagger.
+         */
+        public transient List<Itinerary> itineraries = new ArrayList<>();
```

Comment:
Does this need to be public?

Reply:
This needs to be public for the tests in `ItineraryUtilsTest`.
Swagger will publish all fields in a type regardless of scope according to https://github.com/manusant/spark-swagger/blob/master/src/main/java/io/github/manusant/ss/factory/DefinitionsFactory.java#L36, so we can either leave it listed in the swagger doc for future use, or remove it during post-processing of the generated swagger. Which one should we do?

Reply:
I added code to remove these fields in the public docs in https://github.com/ibi-group/otp-middleware/pull/77/commits/d26baf619770dce9e11667961912b80261de090c.